src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -623,6 +642,56 @@ def __print_restore_preview(kvs_to_restore, kvs_to_modify, kvs_to_delete):
     return True
 
 
+def __is_json_content_type(content_type):
+    # pylint: disable=too-many-return-statements
+    if not content_type:
+        return False
+
+    content_type = content_type.strip().lower()
+    mime_type = content_type.split(';')[0].strip() if ';' in content_type else content_type
+    type_parts = mime_type.split('/') if '/' in mime_type else None
+
+    if not type_parts or len(type_parts) != 2:
+        return False
+
+    (main_type, sub_type) = type_parts
+    if main_type != "application":
+        return False
+    if "json" not in sub_type:
+        return False
+
+    sub_types = sub_type.split('+') if '+' in sub_type else None
+
+    # if sub_types list is not empty, one of the sub_types should exactly match json
+    if sub_types:
+        if "json" in sub_types:
+            return True
+    # if sub_types list is empty, sub_type should exactly match json
+    elif sub_type == "json":
+        return True
+
+    return False
+
+
+def __flatten_json_key_value(key, value, flattened_data, depth, separator):
+    if depth > 1:
+        depth = depth - 1
+        if value and isinstance(value, dict):
+            if separator is None or not separator:
+                raise CLIError(
+                    "A non-empty separator is required for importing hierarchical configurations.")
```

Comment:
Do we fail if no separator specified with nested json? Or we can by default import with first level.

Reply:
Separator is not required by default now. If its not provided, we will default depth to 1.

---
In reply to: [448789846](https://github.com/Azure/azure-cli/pull/14170#discussion_r448789846) [](ancestors = 448789846)